### DISCUSSION TITLE
Can SignalR client use service discovery?

### DISCUSSION THREAD

**Discussion by VincentH-Net** [microsoft/aspire#1075] [Q&A]
I'd like to deploy a (Blazor) chat client app together with its ASP.NET chat service in an Aspire app.

AppHost project:
```csharp
var service = builder.AddProject<Projects.Service>("service");
builder.AddProject<Projects.BlazorSignalRApp>("blazorsignalrapp").WithReference(service);
```
Service project:
```csharp
app.MapHub<ChatHub>("/chathub", options => {
    options.AllowStatefulReconnects = true;
});
```
Blazor project:
```csharp
hubConnection = new HubConnectionBuilder()
.WithUrl("http://service/chathub"
```

All projects are Aspire enabled and call `AddServiceDefaults()`, which should configure all HttpClients with service discovery, however the URL for the SignalR client does not seem to be modified by service discovery; on navigating to the blazor application from the Aspire dashboard, it throws:
```html
HttpRequestException: No such host is known. (service:80)
```

(How) can Aspire Service Discovery be used for the URL of an ASP.NET Core SignalR Client?

**Comment by mohdali** [comment]
I have the same query and currently can't find any reference to help with this.

**Comment by davidfowl** [comment]
See https://github.com/dotnet/aspire/issues/1356
